File: Programming_Languages/Java/Bubble_Sort.java
import java.util.Scanner;
public class Bubble_Sort{  
    static void bubbleSort(int[] arr) {  
        int n = arr.length;  
        int temp = 0;  
         for(int i=0; i < n; i++){  
                 for(int j=1; j < (n-i); j++){  
                          if(arr[j-1] > arr[j]){  
                                 //swap elements  
                                 temp = arr[j-1];  
                                 arr[j-1] = arr[j];  
                                 arr[j] = temp;  
                         }  
                          
                 }  
         }  
  
    }  
    public static void main(String[] args) {  
                System.out.println("Enter number of elements to be Sorted: "); 
                Scanner s=new Scanner(System.in);
		int n = s.nextInt();
		int arr[] = new int[n];

                System.out.println("Enter elements to be sorted: ");
                for(int i=0; i < n; i++){  
                        arr[i] = s.nextInt(); 
                } 
                System.out.println("Array Before Bubble Sort");  
                for(int i=0; i < n; i++){  
                        System.out.print(arr[i] + " ");  
                }  
                System.out.println();  
                  
                bubbleSort(arr);//sorting array elements using bubble sort  
                 
                System.out.println("Array After Bubble Sort");  
                for(int i=0; i < n; i++){  
                        System.out.print(arr[i] + " ");  
                }  
                s.close();
        }  
        
}  

File: Programming_Languages/Javascript/bubblesort.js
const arr = [6, 5, 1, 3]


function bubbleSort(arr) {


    let len = arr.length;
    let swap;

    do {
        swap = false;
        for (let j = 0; j < len - 1; j++) {
            //swap

            if (arr[j] > arr[j + 1]) {
                let temp = arr[j];
                arr[j] = arr[j + 1];
                arr[j + 1] = temp;
                swap = true;
            }
            console.log(j);
        }

    } while (swap);

    // for (let i = 0; i < len; i++) {

    //     for (let j = 0; j < len - 1; j++) {
    //         //swap

    //         if (arr[j] > arr[j + 1]) {
    //             let temp = arr[j];
    //             arr[j] = arr[j + 1];
    //             arr[j + 1] = temp;
    //         }
    //     }

    // }

    return arr;


}

console.log(bubbleSort(arr))
